emcee/ptsampler.py

```diff
@@ -465,16 +468,16 @@ def thermodynamic_integration_log_evidence(self, logls=None, fburnin=0.1):
             return self.thermodynamic_integration_log_evidence(
                 logls=self.lnlikelihood, fburnin=fburnin)
         else:
-            betas = np.concatenate((self.betas, np.array([0])))
-            betas2 = np.concatenate((self.betas[::2], np.array([0])))
+            betas = self.betas
+            betas2 = self.betas[::2]
 
             istart = int(logls.shape[2] * fburnin + 0.5)
 
             mean_logls = np.mean(np.mean(logls, axis=1)[:, istart:], axis=1)
             mean_logls2 = mean_logls[::2]
 
-            lnZ = -np.dot(mean_logls, np.diff(betas))
-            lnZ2 = -np.dot(mean_logls2, np.diff(betas2))
+            lnZ = np.trapz(mean_logls, betas)
```

Comment:
I checked it in the tests, and the "-" is inappropriate here.

On Fri, May 22, 2015 at 1:39 PM Gregory Ashton <notifications@github.com>
wrote:

> In emcee/ptsampler.py
> <https://github.com/dfm/emcee/pull/154#discussion_r30891003>:
>
> >
> >              istart = int(logls.shape[2] * fburnin + 0.5)
> >
> >              mean_logls = np.mean(np.mean(logls, axis=1)[:, istart:], axis=1)
> >              mean_logls2 = mean_logls[::2]
> >
> > -            lnZ = -np.dot(mean_logls, np.diff(betas))
> > -            lnZ2 = -np.dot(mean_logls2, np.diff(betas2))
> > +            lnZ = np.trapz(mean_logls, betas)
>
> Are you not missing the - here?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dfm/emcee/pull/154/files#r30891003>.
>